applications/matter_bridge/src/bridge_shell.cpp

```diff
@@ -46,6 +66,83 @@ static int add_bridged_device(const struct shell *shell, size_t argc, char **arg
 	} else {
 		shell_fprintf(shell, SHELL_ERROR, "Error: internal\n");
 	}
+}
+
+#ifdef CONFIG_BRIDGED_DEVICE_BLE
+static void bluetooth_device_connected(BLEBridgedDevice *device, bt_gatt_dm *discoveredData, bool discoverySucceeded,
+				       void *context)
+{
+	if (context && device && discoveredData && discoverySucceeded) {
+		BluetoothConnectionContext *ctx = reinterpret_cast<BluetoothConnectionContext *>(context);
+
+		ctx->provider->MatchBleDevice(device);
+		if (ctx->provider->ParseDiscoveredData(discoveredData)) {
+			return;
+		}
+
+		add_device(ctx->shell, ctx->deviceType, ctx->nodeLabel, ctx->provider);
+		chip::Platform::Delete(ctx);
+	}
+}
+#endif /* CONFIG_BRIDGED_DEVICE_BLE */
+
+static int add_bridged_device(const struct shell *shell, size_t argc, char **argv)
+{
+	int deviceType = strtoul(argv[1], NULL, 0);
+	char *nodeLabel = nullptr;
+
+#ifdef CONFIG_BRIDGED_DEVICE_BLE
+	int bleDeviceIndex = strtoul(argv[2], NULL, 0);
+
+	if (argv[3]) {
+		nodeLabel = argv[3];
+	}
+
+	/* The device object can be created once the Bluetooth LE connection will be established. */
+	BluetoothConnectionContext *context = chip::Platform::New<BluetoothConnectionContext>();
+
+	if (!context) {
+		return -ENOMEM;
+	}
+
+	Platform::UniquePtr<BluetoothConnectionContext> contextPtr(context);
+	contextPtr->shell = shell;
+	contextPtr->deviceType = deviceType;
+
+	if (nodeLabel) {
+		strcpy(contextPtr->nodeLabel, nodeLabel);
+	}
+
+	BridgedDeviceDataProvider *provider = create_ble_provider(contextPtr->deviceType);
+
+	if (!provider) {
+		return -ENOMEM;
+	}
+
+	contextPtr->provider = reinterpret_cast<BLEBridgedDeviceProvider *>(provider);
+	CHIP_ERROR err = GetBLEConnectivityManager().Connect(bleDeviceIndex, bluetooth_device_connected, contextPtr->get(),
```

Comment:
We are passing the context raw pointer to the ConnectivityManager::Connect and we release the uniqe_ptr if the operation succeeded, which is good. But then which entity is responsible for managing the context's life time? I would expect the ConnectivityManager, but eventually we land in `bluetooth_device_connected` callback, right? Should the context pointer be released there then?

Reply:
I think you put this comment without refreshing, as it was fixed earlier. The unique ptr is created in connected callback, so the context should be deleted properly.

Reply:
Right.